### PR TITLE
feat/decouple back button

### PR DESCRIPTION
**Pull request by manansharma18**
Screenshot 2023-11-05 at 11 58 40 AM

https://github.com/player-ui/player/issues/169
### Change Type (required)
Indicate the type of change your pull request is:
Adds a role property inside the metaData on the action asset. This property is used to decouple the back Icon from the back button.

- [X] `patch`
- [ ] `minor`
- [ ] `major`

### CODE REVIEW

File: plugins/reference-assets/core/src/assets/action/types.ts [hunk 1]
```diff
@@ -27,6 +27,9 @@ export interface ActionAsset<AnyTextAsset extends Asset = Asset>
 
     /** Force transition to the next view without checking for validation */
     skipValidation?: boolean;
+
+    /** boolean value to decide for the left anchor sign */
+    role?: boolean;
```
Comment (hborawski) [inline plugins/reference-assets/core/src/assets/action/types.ts:32]: I think we would want this to be `role?: string;` so we can potentially support other roles in the future, but the transform can populate it with a `back` value when appropriate

File: plugins/reference-assets/mocks/action/action-navigation.json [hunk 2]
```diff
@@ -26,6 +26,23 @@
...
+            "value": "Prev",
+            "label": {
+              "asset": {
+                "id": "action-prev-id-without-icon",
+                "type": "text",
+                "value": "Go Back Without Icon"
+              }
+            },
+              "metaData": {
+                "role": false
```
Comment (hborawski) [inline plugins/reference-assets/mocks/action/action-navigation.json:42]: Is this to test that the transform will set it to `true`? 

If this is set explicitly, would we even want to change it?

**Review by hborawski — CHANGES_REQUESTED** [review]

File: plugins/reference-assets/core/src/assets/action/types.ts (hunk 1 above)
Comment (manansharma18) [reply plugins/reference-assets/core/src/assets/action/types.ts:32]: Hey @hborawski , so if I make the role as string then In the transform, I need to check if it is a backButton and then apply the role as 'back'(maybe). In the action component, then check for the role value. Is this understanding correct?

File: plugins/reference-assets/mocks/action/action-navigation.json (hunk 2 above)
Comment (manansharma18) [reply plugins/reference-assets/mocks/action/action-navigation.json:42]: So, currently the user can specify the role property as true/false to show/hide the back icon. This part of story book shows a back button with a back icon

File: plugins/reference-assets/core/src/assets/action/types.ts (hunk 1 above)
Comment (hborawski) [reply plugins/reference-assets/core/src/assets/action/types.ts:32]: yes this is correct, as `"role": true` does not really have any semantic intent, `"role": "back"` is more clear as to what is happening,

File: plugins/reference-assets/mocks/action/action-navigation.json (hunk 2 above)
Comment (hborawski) [reply plugins/reference-assets/mocks/action/action-navigation.json:42]: if this is `false`, why would this show the icon?

Comment (manansharma18) [reply plugins/reference-assets/mocks/action/action-navigation.json]: Hey, this frf is to show, the useage of `'role':'back'`. The one above this shows how to get the button without the back icon.

File: plugins/reference-assets/core/src/assets/action/types.ts [hunk 3]
```diff
@@ -27,6 +27,9 @@ export interface ActionAsset<AnyTextAsset extends Asset = Asset>
 
     /** Force transition to the next view without checking for validation */
     skipValidation?: boolean;
+
+    /** boolean value to decide for the left anchor sign */
```
Comment (hborawski) [inline plugins/reference-assets/core/src/assets/action/types.ts:31]: this docstring should be updated since it is a string now

**Review by hborawski — APPROVED** [review]

**Review by mercillo — APPROVED** [review]

**Pull request merged**